src/keychain/memory.rs

```diff
@@ -0,0 +1,78 @@
+use crate::{Height, Keychain};
+
+use fuel_crypto::borrown::Borrown;
+use fuel_crypto::{Hasher, Keystore, SecretKey, Signer};
+use rand::rngs::StdRng;
+use rand::SeedableRng;
+
+use core::convert::Infallible;
+use core::ops::{Range, RangeBounds};
+use std::collections::HashMap;
+
+/// Default in-memory implementation of a keychain
+#[derive(Debug, Default, Clone)]
+pub struct MemoryKeychain {
+    keys: HashMap<Range<Height>, SecretKey>,
```

Comment:
Indeed, what you said reflect what we discussed during ethconnect. However, this PR was written before that. To avoid further increase this already gigantic PR, I think its better we introduce these new concepts in a new one

Reply:
makes sense